app/src/main/java/com/example/rickmortymvvm/app/util/TimberLoggingInterceptor.kt

```diff
@@ -0,0 +1,45 @@
+package com.example.rickmortymvvm.app.util
+
+import okhttp3.Interceptor
+import okhttp3.Request
+import okhttp3.Response
+import okhttp3.ResponseBody
+import okio.Buffer
+import timber.log.Timber
+import java.io.IOException
+
+class TimberLoggingInterceptor : Interceptor {
+
+    override fun intercept(chain: Interceptor.Chain): Response {
+        val request = chain.request()
+
+        val t1 = System.nanoTime()
+        Timber.i("Sending request %s on %s%n%s", request.url(), chain.connection(), request.headers())
+        Timber.v("REQUEST BODY BEGIN\n%s\nREQUEST BODY END", bodyToString(request))
+
+        val response = chain.proceed(request)
+
+        val responseBody: ResponseBody? = response.body()
+        val responseBodyString = response.body()?.string()
+
+        val newResponse = response.newBuilder().body(ResponseBody.create(responseBody?.contentType(), responseBodyString?.toByteArray())).build()
```

Comment:
Quando a linha fica muito comprida... Complica a leitura/entendimento do codigo... Ve se essa sugestão não deixa mais claro o que esta acontecendo..
```suggestion
        val newResponse = response.newBuilder()
             .body(ResponseBody.create(
                  responseBody?.contentType(),
                  responseBodyString?.toByteArray()
             )).build()
```